game/plugins/src/main/kotlin/gg/rsmod/plugins/content/skills/thieving/pickpocket/pickpocket.kts

```diff
@@ -0,0 +1,64 @@
+package gg.rsmod.plugins.content.skills.thieving.pickpocketing
+
+import gg.rsmod.plugins.content.combat.isAttacking
+import gg.rsmod.plugins.content.combat.isBeingAttacked
+import gg.rsmod.plugins.content.skills.thieving.pickpocket.PickpocketNpcs
+
+private val PICKPOCKET_ANIMATION = 881
+private val GLOVES_OF_SILENCE_BONUS = 5
+
+PickpocketNpcs.values.forEach { pickpocketNpc ->
+    pickpocketNpc.npcIds.forEach { npcId ->
+        on_npc_option(npc = npcId, option = "pickpocket") {
+            player.queue {
+                pickpocket(npcId, pickpocketNpc)
+            }
+        }
+    }
+}
+
+suspend fun QueueTask.pickpocket(npcId: Int, npcType: PickpocketNpcs) {
+    val thievLvl: Int = player.getSkills().getCurrentLevel(Skills.THIEVING)
+    val npcName = npcType.npcName ?: world.definitions.get(NpcDef::class.java, npcId).name
+    if (thievLvl < npcType.lvl) {
+        player.message("You need level ${npcType.lvl} thieving to pick the $npcName's pocket.")
+        return
+    }
+    if (player.isAttacking() || player.isBeingAttacked()) {
+        player.message("You can't pickpocket while in combat!")
+        return
+    }
+    if (!player.inventory.hasSpace) {
+        player.message("You don't have enough inventory space to pickpocket!")
+        return
+    }
+
+    //pickpocketing animation and starting message
+    player.animate(PICKPOCKET_ANIMATION)
+    player.message("You attempt to pickpocket the $npcName...")
+
+    //wait 3 game cycles
+    player.lock = LockState.FULL_WITH_ITEM_INTERACTION
+    wait(3)
+    player.lock = LockState.NONE
+
+    //determine if the pickpocket was successful or not by "if random number is within success chances"
+    val bonus = if (player.hasEquipped(EquipmentType.GLOVES, Items.GLOVES_OF_SILENCE)) GLOVES_OF_SILENCE_BONUS else 0
+    if (thievLvl.interpolate(55, 95, npcType.lvl, 99, 100 - bonus)) {
```

Comment:
Personally I would probably separate this logic and have it in a separate method/location as it is pretty important. That way it can be documented better and not just discretely shoved in like this.